airbyte-integrations/connectors/source-hubspot/source_hubspot/components.py

```diff
@@ -0,0 +1,43 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, Dict, Optional
+
+from airbyte_cdk.sources.declarative.transformations import RecordTransformation
+from airbyte_cdk.sources.types import Config, StreamSlice, StreamState
+
+
+class NewtoLegacyFieldTransformation(RecordTransformation):
+    """
+    Implements a custom transformation which adds the legacy field equivalent of v2 fields for streams which contain Deals and Contacts entities.
```

Comment:
only nit is that can we include an explanation why the existing `AddFields` component was not compatible with hubspot to justify the need for a custom implementation. Hopefully our future selves thank us

Reply:
Added. Good callout.